src/tiled/objectsdock.cpp

```diff
@@ -342,7 +342,7 @@ void ObjectsView::mousePressEvent(QMouseEvent *event)
             DocumentManager::instance()->centerMapViewOn(center + offset);
         }
     } else if (Layer *layer = mapObjectModel()->toLayer(index)) {
-        mMapDocument->setCurrentObject(layer);
+        mMapDocument->setCurrentLayer(layer);
```

Comment:
Please keep the call to `setCurrentObject`, since changing the current layer will only automatically change the current object when either there's isn't a current object or it is a layer. But when the user clicks a layer he may expect to see its properties, for which we need to change the current object.

In any case I agree it would be expected that the current layer changes as well. I do not remember why this code was not doing that in the first place. Maybe there was no good reason, or maybe we'll find out later. :-)